sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2927,6 +2927,17 @@ object SQLConf {
       // show full stacktrace in tests but hide in production by default.
       .createWithDefault(Utils.isTesting)
 
+  val PYTHON_UDF_PROFILER =
+    buildConf("spark.sql.pyspark.udf.profiler")
+      .doc("Configure the Python/Pandas UDF profiler by enabling or disabling it " +
+        "with the option to choose between \"perf\" and \"memory\" types, " +
+        "or unsetting the config disables the profiler. This is disabled by default.")
+      .version("4.0.0")
+      .stringConf
+      .transform(_.toLowerCase(Locale.ROOT))
+      .checkValues(Set("perf", "memory"))
```

Comment:
I wonder if it's more straightforward to use the module name. e.g., `cProfile` and `memory-profiler`

Reply:
I noticed there are multiple user-facing references to the current "perf" profiler: [Python Profilers for UDFs ](https://spark.apache.org/docs/latest/api/python/development/debugging.html#id7), [Workers profiling](https://www.databricks.com/blog/how-profile-pyspark). It would be great we could make them consistent.

Reply:
@xinrong-meng what's the suggestion? could you elaborate?

Reply:
I could adjust those references once we decide a standard name.